src/AutoRest.CSharp/Common/Generation/Writers/FormattableStringHelpers.cs

```diff
@@ -85,6 +85,11 @@ public static FormattableString GetIdentifiersFormattable(this IEnumerable<strin
 
         public static FormattableString GetConversionFormattable(this Parameter parameter, CSharpType toType)
         {
+            if (TypeFactory.IsDictionary(parameter.Type) && toType.EqualsIgnoreNullable(typeof(RequestContent)))
```

Comment:
Since our helper method does not accept `IReadOnlyDictionary`, could we change the condition to `IsReadWriteDictionary`?